inst/templates/bs4_book.html

```diff
@@ -10,6 +10,17 @@
   <meta name="author" content="$author-meta$" />
   $endfor$
 
+  <meta property="og:title" content="$pagetitle$" />
+  <meta property="og:type" content="book" />
+  $if(url)$<meta property="og:url" content="$url$" />$endif$
+  $if(cover-image)$<meta property="og:image" content="$url$/$cover-image$" />$endif$
+  $if(description)$<meta property="og:description" content="$description$" />$endif$
```

Comment:
I think I deliberately omitted this because I don't like how every chapter generates exactly the same preview (which I think (e.g.) generates confusing tweets). So if we're to turn it back on, I'd like to provide someway to customise it per chapter. Do you have any thoughts on where to do that? Presumably we'd want to match existing Rmd/pkgdown conventions.

Reply:
I don't mind the image being the same across chapters within the same book. 

The metadata on the other hand would be nice to have vary across chapter. For a Hugo theme, I like to put the site title/section title first then the page, so in bookdown terms maybe the title then the chapter name? I'm not sure if you can add individual chapter descriptions though with no YAML 🤔 
<img width="1080" alt="Screen Shot 2021-01-04 at 12 21 33 PM" src="https://user-images.githubusercontent.com/12160301/103561509-9e667b80-4e87-11eb-9d6e-a821aa2fc03e.png">

Reply:
Also want to link to metathis, which allows you to insert metadata in an R chunk: https://github.com/gadenbuie/metathis- could work within a chapter?

Reply:
There was an issue open re. same metadata for each page: https://github.com/rstudio/bookdown/issues/970
It seems it mislead open graph scraper when the og:url is set to the main page instead of the current page, because og:url is expected to be the "canonical URL of the page" not the website. @hadley did you have that in mind too ? 

This could be a good opportunity to do it correctly or offer a way to make this configurable.

Reply:
@cderv good point — maybe it's worth generating the HTML from R, and then that function can be used from `bs4_book()` and `gitbook()`.

Reply:
Will probably be a bit easier to do the metadata for `bs4_book()` since there's a one-to-one correspondence between Rmd and html.